### DISCUSSION TITLE
Is Ariadne Codegen library intended for use as part of ongoing development?

### DISCUSSION THREAD

**Discussion by steve-marmalade** [mirumee/ariadne#1038] [General]
Apologies if this question should have be asked in a different forum, but wondering about the intended use of this library. It looks primarily designed to bootstrap an entire backend server based on a GraphQL schema (which is neat). I wonder if you also intend for it to be used as a more modest utility that keeps a file (or files) of generated pydantic dataclasses/models in sync with the GraphQL schema.

So the workflow would be:
- Modify the GraphQL schema
- Run the utility
- Reference any newly generated models in resolvers (or immediately get the benefit of having existing models updated)

Perhaps the contract would be: each `*.graphql` file in the repo gets it's own `*.py` companion with the relevant type definitions.

**Comment by rafalp** [comment]
Issues are only for tracking the roadmap and bugs. We ask in our readme that usage questions be raised on [Ariadne's discussions](https://github.com/mirumee/ariadne/discussions/) :)

Edit: Woah, GitHub let me do it myself :D

**Comment by rafalp** [comment]
> It looks primarily designed to bootstrap an entire backend server based on a GraphQL schema (which is neat)

This is not what Ariadne Codegen does. Instead it creates Python package implementing the **GraphQL client** for a given schema and GraphQL queries.

**Reply by steve-marmalade** [reply]
Got it - thank you for the clarification
